### ISSUE TITLE
potential bug report on visitChildren() in c++ in ANTLR 4.9.3

### ISSUE THREAD

**Issue by chakpongchung** [antlr/antlr4#3512]
- [x] I have reproduced my issue using the latest version of ANTLR
- [x] I have submitted my issue to [antlr4-discussion google group](https://groups.google.com/forum/#!forum/antlr-discussion) or asked at [stackoverflow](http://stackoverflow.com/questions/tagged/antlr4)
- [ ] Responses from the above seem to indicate that my issue could be an ANTLR bug
- [x] I have done a search of the existing issues to make sure I'm not sending in a duplicate

 - target information 
C++

 - smallest possible grammar and code that reproduces the behavior
 
A minimal code repo to reproduce the problem:

https://github.com/chakpongchung/debugging

 - description of the expected behavior and actual behavior: 
 I expect to see the result from a child node propagating to the parent node. Currently, I have to specify which visit function to use on specific children to see the propagated result:
 
 ```
         auto tmp1 = visitOC_PropertyOrLabelsExpression(
                ctx->oC_PropertyOrLabelsExpression());
        cout << "tmp1: " << tmp1.is<std::string>() << endl;
 ```
 from https://github.com/chakpongchung/debugging/blob/main/CypherCustomVisitor.h#L18

I suspect the visitChildren() in C++ might have problems. I tried using it the same way in older version of ANTLR4 or even a similar way in Java, it works well.

The problem is here:
https://github.com/chakpongchung/debugging/blob/main/CypherCustomVisitor.h#L25

**Comment by chakpongchung** [comment]
@jcking @parrt @mike-lischke

**Comment by chakpongchung** [comment]
fixed in next release(>4.9.3)
